src/pretix/plugins/paypal2/payment.py

```diff
@@ -0,0 +1,1003 @@
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations under the License.
+import hashlib
+import json
+import logging
+import urllib.parse
+from collections import OrderedDict
+from decimal import Decimal
+
+from django import forms
+from django.contrib import messages
+from django.http import HttpRequest
+from django.template.loader import get_template
+from django.templatetags.static import static
+from django.urls import reverse
+from django.utils.crypto import get_random_string
+from django.utils.safestring import mark_safe
+from django.utils.timezone import now
+from django.utils.translation import gettext as __, gettext_lazy as _
+from django_countries import countries
+from i18nfield.strings import LazyI18nString
+from paypalcheckoutsdk.orders import (
+    OrdersCaptureRequest, OrdersCreateRequest, OrdersGetRequest,
+    OrdersPatchRequest,
+)
+from paypalcheckoutsdk.payments import CapturesRefundRequest, RefundsGetRequest
+
+from pretix import settings
+from pretix.base.decimal import round_decimal
+from pretix.base.forms.questions import guess_country
+from pretix.base.models import Event, Order, OrderPayment, OrderRefund, Quota
+from pretix.base.payment import BasePaymentProvider, PaymentException
+from pretix.base.services.mail import SendMailException
+from pretix.base.settings import SettingsSandbox
+from pretix.helpers.urls import build_absolute_uri as build_global_uri
+from pretix.multidomain.urlreverse import build_absolute_uri, eventreverse
+from pretix.plugins.paypal2.client.core.environment import (
+    LiveEnvironment, SandboxEnvironment,
+)
+from pretix.plugins.paypal2.client.core.paypal_http_client import (
+    PayPalHttpClient,
+)
+from pretix.plugins.paypal2.client.customer.partner_referral_create_request import (
+    PartnerReferralCreateRequest,
+)
+from pretix.plugins.paypal.models import ReferencedPayPalObject
+
+logger = logging.getLogger('pretix.plugins.paypal')
+
+SUPPORTED_CURRENCIES = ['AUD', 'BRL', 'CAD', 'CZK', 'DKK', 'EUR', 'HKD', 'HUF', 'INR', 'ILS', 'JPY', 'MYR', 'MXN',
+                        'TWD', 'NZD', 'NOK', 'PHP', 'PLN', 'GBP', 'RUB', 'SGD', 'SEK', 'CHF', 'THB', 'USD']
+
+LOCAL_ONLY_CURRENCIES = ['INR']
+
+
+class PaypalSettingsHolder(BasePaymentProvider):
+    identifier = 'paypal_settings'
+    verbose_name = _('PayPal')
+    is_enabled = False
+    is_meta = True
+    payment_form_fields = OrderedDict([])
+
+    def __init__(self, event: Event):
+        super().__init__(event)
+        self.settings = SettingsSandbox('payment', 'paypal', event)
+
+    @property
+    def settings_form_fields(self):
+        # PayPal connect (legacy) || ISU
+        if self.settings.connect_client_id and self.settings.connect_secret_key and not self.settings.secret:
+            if self.settings.connect_user_id:
```

Comment:
Is there any possible way to have PPv2 active and still have a connect_user_id? Probably not, and also wouldn't work.